app/services/solidus_square/create_payment_service.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module SolidusSquare
+  class CreatePaymentService < Base
+    attr_reader :source_id, :order, :payment_method_id
+
+    def initialize(source_id:, order:, payment_method_id:)
+      @source_id = source_id
+      @order = order
+      @payment_method_id = payment_method_id
+
+      super()
+    end
+
+    def call
+      order.next! if order.delivery?
+      complete_payment!
+    end
+
+    private
+
+    def complete_payment!
+      payment = create_payment!
+      payment.complete! if payment.source.captured?
+    end
+
+    def square_order_id
+      square_payment_response[:order_id]
+    end
+
+    def square_payment_response
+      @square_payment_response ||= gateway.create_payment(square_order_total, source_id,
+        payment_method.auto_capture?)
+    end
+
+    def create_payment!
+      order.payments.find_or_create_by!(response_code: square_order_id) do |payment|
+        payment.amount = payment_amount
+        payment.source = ::SolidusSquare::PaymentSource.create!(construct_payment_source)
+        payment.payment_method_id = payment_method_id
+        payment.state = 'pending'
+      end
+    end
+
+    def payment_method
+      @payment_method ||= ::Spree::PaymentMethod.find(payment_method_id)
+    end
+
+    def gateway
+      @gateway ||= payment_method.gateway
+    end
+
+    def payment_amount
+      square_payment_response[:amount_money][:amount] / 100.0
+    end
+
+    def square_order_total
+      (order.total * 100).to_i
+    end
```

Comment:
What do you think to move the conversion from decimal to cents into the `SolidusSquare::Payments::Create`

Reply:
https://github.com/nebulab/solidus_square/pull/49/commits/47433f81b4e5d4331e2393b0dbf810e3f92b3bb3